src/main/java/gregtech/common/covers/detector/CoverDetectorFluidAdvanced.java

```diff
@@ -0,0 +1,172 @@
+package gregtech.common.covers.detector;
+
+import codechicken.lib.raytracer.CuboidRayTraceResult;
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.CoverWithUI;
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.util.GTLog;
+import gregtech.api.util.GTUtility;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.common.covers.filter.FluidFilterContainer;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.*;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+
+import java.util.regex.Pattern;
+
+public class CoverDetectorFluidAdvanced extends CoverDetectorFluid implements CoverWithUI {
+
+    private static final int PADDING = 3;
+    private static final int SIZE = 18;
+
+    private static final int DEFAULT_MIN = 1000; // 1 Bucket
+    private static final int DEFAULT_MAX = 16000; // 16 Buckets
+
+    private int min, max;
+
+    protected FluidFilterContainer fluidFilter;
+
+    public CoverDetectorFluidAdvanced(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        this.fluidFilter = new FluidFilterContainer(this);
+        this.min = DEFAULT_MIN;
+        this.max = DEFAULT_MAX;
+    }
+
+    @Override
+    public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, CuboidRayTraceResult hitResult) {
+        if (!this.coverHolder.getWorld().isRemote) {
+            openUI((EntityPlayerMP) playerIn);
+        }
+        return EnumActionResult.SUCCESS;
+    }
+
+    @Override
+    public void renderCover(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline, Cuboid6 plateBox, BlockRenderLayer layer) {
+        // todo replace with unique texture
```

Comment:
Remove todo when it is implemented.
```suggestion
        // todo replace with unique texture
```